Liquid Glass StateButton

DRadmir opened the pull request:
Replaces usage of the .primary button type with .adoptiveGlassEffect across multiple scenes, view models, and tests. Adds adoptiveGlassEffect support to ButtonType and StateButtonStyle, including a glass effect for iOS 26 and above. Updates related tests and disables logic to use the new style, providing a more modern button appearance throughout the app.

White artifacts above the keyboard fixed in PR: https://github.com/gemwalletcom/gem-ios/pull/1170

Close: https://github.com/gemwalletcom/gem-ios/issues/1159

## iOS 18.5
<img width="320" alt="Simulator Screenshot - iPhone 16e - 2025-09-16 at 21 36 49" src="https://github.com/user-attachments/assets/b64c3e80-f291-4dd8-9f2f-f5a48f46306f" />
<img width="320" alt="Simulator Screenshot - iPhone 16e - 2025-09-16 at 21 36 54" src="https://github.com/user-attachments/assets/007e0bd7-f67b-44b2-8ced-e1f7ffbafdaa" />
<img width="320" alt="Simulator Screenshot - iPhone 16e - 2025-09-16 at 21 37 03" src="https://github.com/user-attachments/assets/dab019bd-980b-4983-b760-0dd904feac1a" />
<img width="320" alt="Simulator Screenshot - iPhone 16e - 2025-09-16 at 21 37 08" src="https://github.com/user-attachments/assets/416f4e10-bcf5-415f-b73d-3503c0280ea6" />


## iOS 26
<img width="320" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-16 at 21 38 36" src="https://github.com/user-attachments/assets/43ec6a3c-c38c-40c3-9975-07d54772b5e8" />
<img width="320" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-16 at 21 39 11" src="https://github.com/user-attachments/assets/9de8f341-28ee-404d-82ae-74b2aa7613ad" />
<img width="320" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-16 at 21 39 19" src="https://github.com/user-attachments/assets/80add232-321f-4525-ad0d-d81b2ab60ad2" />
<img width="320" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-16 at 21 39 26" src="https://github.com/user-attachments/assets/b68d38da-0d4a-4d8b-8699-d76fa3fa805e" />


https://github.com/user-attachments/assets/eb317762-a585-4fd8-8231-5b2d2a572c28


https://github.com/user-attachments/assets/43739a44-6dd3-401e-975a-00484c543c30

DRadmir committed:
Switch StateButton to adoptiveGlassEffect style

Replaces usage of the .primary button type with .adoptiveGlassEffect across multiple scenes, view models, and tests. Adds adoptiveGlassEffect support to ButtonType and StateButtonStyle, including a glass effect for iOS 26 and above. Updates related tests and disables logic to use the new style, providing a more modern button appearance throughout the app.

DRadmir committed:
Add glass effect option to ColorButtonStyle

Introduces an isGlassEffectEnabled parameter to ColorButtonStyle, allowing buttons to use a glass effect background on supported iOS versions. Updates all static constructors to support the new parameter. Also simplifies the glassEffect usage in StateButtonStyle to use only the interactive state.

gemcoder21 commented on Features/Assets/Sources/Scenes/AddTokenScene.swift:
```diff
@@ -32,7 +32,7 @@ public struct AddTokenScene: View {
             StateButton(
                 text: model.actionButtonTitle,
-                type: .primary(model.state),
+                type: .adoptiveGlassEffect(model.state),
```
```suggestion
                type: .primary(model.state),
```

we should make this primary, the rest happens behind the scenes

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1159-adjust-action-button-to-liquid-glass

# Conflicts:
#	Features/Assets/Sources/Scenes/AddTokenScene.swift

DRadmir committed:
Remove adoptiveGlassEffect button type in favor of primary

Replaces all usages of ButtonType.adoptiveGlassEffect with ButtonType.primary across the codebase. Removes the adoptiveGlassEffect implementation and related tests, consolidating button styling logic to use the primary variant. This simplifies button type management and reduces code duplication.

gemcoder21 commented on Packages/Components/Tests/ComponentsTests/ButtonTypeTests.swift:
```diff
@@ -36,4 +36,4 @@ struct ButtonTypeTests {
 
-private struct TestError: Error {}
\ No newline at end of file
+private struct TestError: Error {}
```
can we remove it?

DRadmir replied:
Used for testing 
`#expect(ButtonType.primary(StateViewType<String>.error(TestError())) == .primary(.disabled))`
